### PR TITLE
feat(bridge): Enhance `webhook` settings to increase usability

### LINKED ISSUE #8652: Explain webhook started / finished as async or sync
Improve / add to the wording of the webhok service screen so tell users that the model is sync or async.

**Describe the solution you'd like**
Adjust the wording on the webhook bridge screen to explain started and finished events more clearly to the user.

Sending started and finished from webhook service is an "async" model. The Keptn task immediately returns. In contrast, delegating the finished event to the receiving service will wait until the receiver is done. Thus it is a synchronous model and will "block.

Async is easier at first to implement as it needs no changes to the receiving side, but the task timing does not include the processing time of the receiving system. Sync allows users to send data from the system back to Keptn and gives accurate timings, but requires modifications.

Users usually start with the async model as it is easiest to implement.

image

### PR DESCRIPTION
**Pull request by afzal442**
Signed-off-by: afzal442 <afzal442@gmail.com>

## This PR

allows user to be able to clarify webhook settings
- adds this new feature

### Related Issues

Fixes #8652

### CODE REVIEW

**Comment by afzal442** [comment]
hey @agardnerIT ! Could you review that? Thanks

**Review by mowies — CHANGES_REQUESTED** [review]
Please remove the added submodule from the PR

**Review by agardnerIT — CHANGES_REQUESTED** [review]
Errant / missing quote: `"block`. In `#sendFinishedOverlay` perhaps add words to make it absolutely clear that the users would be responsible for crafting and sending a `.finished` event back from the tool via HTTPS POST.

As regular Keptn users, we know this but a first time user may miss that subtle message and expect it to "magically happen"

**Comment by afzal442** [comment]
hmm! That's right. It looks duplicated here. Any suggestion plz? If I understand this ```.finished event back from the tool via HTTPS POST``` correct is keptn will receive events from the integration tool after it finishes the job via http POST request.

File: bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html [hunk 1]
```diff
@@ -184,11 +184,15 @@ <h2>Webhook configuration</h2>
...
-        If you subscribe your webhook to a task event of type triggered, the corresponding started and finished events
-        are sent automatically.
+        Sending started and finished from webhook service is an "async" model. The Keptn task immediately returns. Async
+        is easier at first to implement as it needs no changes to the receiving side, but the task timing does not
+        include the processing time of the receiving system. If you subscribe your webhook to a task event of type
+        triggered, the corresponding started and finished events are sent automatically.
         <br /><br />
-        To move the responsibility of sending the started event to the webhook receiver, select the option 'by webhook
-        receiver'.
+        Delegating the finished event to the receiving service will wait until the receiver is done. Thus it is a
```
Comment (johannes-b) [inline bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html:192]: Could you please change this paragraph to: 

Delegating the stated event to the receiving service will block the task execution until the receiving service sends the started event. However, this allows users to send data from the system back to Keptn and gives accurate timings. To move the responsibility of sending the started event to the webhook receiver, select the option 'by webhook receiver'.

File: bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html [hunk 2]
```diff
@@ -205,11 +209,16 @@ <h2>Webhook configuration</h2>
...
-        If you subscribe your webhook to a task event of type triggered, the corresponding started and finished events
-        are sent automatically.
+        Sending started and finished from webhook service is an "async" model. The Keptn task immediately returns. Async
+        is easier at first to implement as it needs no changes to the receiving side, but the task timing does not
+        include the processing time of the receiving system. If you subscribe your webhook to a task event of type
+        triggered, the corresponding started and finished events are sent automatically.
         <br /><br />
-        To move the responsibility of sending the finished event to the webhook receiver, select the option 'by webhook
-        receiver'. This is especially required for webhooks that trigger a long-running task.
+        Delegating the finished event to the receiving service will wait until the receiver is done. Thus it is a
```
Comment (johannes-b) [inline bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html:217]: Could you please change this paragraph to: 

Delegating the finished event to the receiving service will block the task execution until the receiving service sends the finished event. However, this allows users to send data from the system back to Keptn and gives accurate timings. To move the responsibility of sending the finished event to the webhook receiver, select the option 'by webhook receiver'.

File: bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html [hunk 3]
```diff
@@ -184,11 +184,15 @@ <h2>Webhook configuration</h2>
     </div>
     <ng-template #sendStartedOverlay>
       <p>
-        If you subscribe your webhook to a task event of type triggered, the corresponding started and finished events
-        are sent automatically.
+        Sending started and finished from webhook service is an "async" model. The Keptn task immediately returns. Async
```
Comment (johannes-b) [inline bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html:187]: I would shorten the paragraph a little bit by focusing on the benefit that the receiver does not need to send the started event. 

My proposal: 
If you subscribe your webhook to a task event of type triggered, the corresponding started event is sent automatically. Consequently, the receiving service does not need to send back the started event.

File: bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html [hunk 4]
```diff
@@ -205,11 +209,16 @@ <h2>Webhook configuration</h2>
     </div>
     <ng-template #sendFinishedOverlay>
       <p>
-        If you subscribe your webhook to a task event of type triggered, the corresponding started and finished events
-        are sent automatically.
+        Sending started and finished from webhook service is an "async" model. The Keptn task immediately returns. Async
```
Comment (johannes-b) [inline bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html:212]: I would shorten the paragraph a little bit by focusing on the benefit that the receiver does not need to send the finished event. 

My proposal: 
If you subscribe your webhook to a task event of type triggered, the corresponding finished event is sent automatically. Consequently, the receiving service does not need to send back the finished event.

**Comment by afzal442** [comment]
Thanks @johannes-b your suggestions make sense to me. I will make change as per that.

File: bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html [hunk 5]
```diff
@@ -184,11 +184,15 @@ <h2>Webhook configuration</h2>
     </div>
     <ng-template #sendStartedOverlay>
       <p>
-        If you subscribe your webhook to a task event of type triggered, the corresponding started and finished events
-        are sent automatically.
+        Sending started and finished from webhook service is an "async" model. The Keptn task immediately returns. Async
+        is easier at first to implement as it needs no changes to the receiving side, but the task timing does not
+        include the processing time of the receiving system. If you subscribe your webhook to a task event of type
+        triggered, the corresponding started and finished events are sent automatically.
```
Comment (StackScribe) [inline bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html:190]:
```suggestion
        ` triggered`, the corresponding started and finished events are sent automatically.
```

Comment (afzal442) [reply bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.html]: Thanks @StackScribe ! I just made changes without noticing you. Could you review again? 😄

**Comment by afzal442** [comment]
Closing as stale and being referred by anyone

cc @thisthat

**Pull request closed without merging**